Don't omit coop maps from the local map list

andreasley opened the pull request:
Fixes #1560 and probably also #1246.

Previously, only skirmish maps were added to the list of installed maps. This led to replays of coop missions not showing up.
This patch fixes this while retaining the filter for skirmish maps when creating a new game.

andreasley committed:
Fixes #1560 by no longer omitting coop maps from the local map list

Geosearchef commented on src/main/java/com/faforever/client/map/MapService.java:
```diff
@@ -303,25 +303,20 @@ public Image loadPreview(String mapName, PreviewSize previewSize) {
+      return Optional.empty();
+    } else {
+      MapBean mapBean = mapsByFolderName.get(mapFolderKey);
+      return Optional.of(mapBean);
```
I'd either remove the else, therefore having it as an abort condition or swap if and else around.

Geosearchef reviewed (dismissed):
Looks good to me, are you sure installedMaps and mapsByName are kept im sync correctly?

bukajsytlos commented:
or return Optional.ofNullable(map. getValue(key))

On Mon, 20 Jan 2020, 14:19 Geosearchef, <notifications@github.com> wrote:

> *@Geosearchef* approved this pull request.
>
> Looks good to me, are you sure installedMaps and mapsByName are kept im
> sync correctly?
> ------------------------------
>
> In src/main/java/com/faforever/client/map/MapService.java
> <https://github.com/FAForever/downlords-faf-client/pull/1561#discussion_r368542373>
> :
>
> >    public Optional<MapBean> getMapLocallyFromName(String mapFolderName) {
>      logger.debug("Trying to find map '{}' locally", mapFolderName);
> -    ObservableList<MapBean> installedMaps = getInstalledMaps();
> -    synchronized (installedMaps) {
> -      for (MapBean mapBean : installedMaps) {
> -        if (mapFolderName.equalsIgnoreCase(mapBean.getFolderName())) {
> -          logger.debug("Found map {} locally", mapFolderName);
> -          return Optional.of(mapBean);
> -        }
> -      }
> +    String mapFolderKey = mapFolderName.toLowerCase();
> +    if (!mapsByFolderName.containsKey(mapFolderKey)) {
> +      return Optional.empty();
> +    } else {
> +      MapBean mapBean = mapsByFolderName.get(mapFolderKey);
> +      return Optional.of(mapBean);
>
> I'd either remove the else, therefore having it as an abort condition or
> swap if and else around.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/FAForever/downlords-faf-client/pull/1561?email_source=notifications&email_token=ACBYJJP2SL5EKMYIM75V56LQ6WQFPA5CNFSM4KI3F7T2YY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCSKN6VY#pullrequestreview-345300823>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACBYJJKGFH7ZE62K5WRUGTLQ6WQFPANCNFSM4KI3F7TQ>
> .
>

andreasley commented:
> Looks good to me, are you sure installedMaps and mapsByName are kept im sync correctly?

Reasonably sure, as there's a change listener on `installedMaps` that updates `mapsByName`.

andreasley committed:
Simplified code

https://github.com/FAForever/downlords-faf-client/pull/1561#discussion_r368542373

1-alex98 reviewed (changes requested):
What about the maps shon in the create game dialog do they now contain coop maps!!???

andreasley commented:
> What about the maps shon in the create game dialog do they now contain coop maps!!???

No, of course not. :)

1-alex98 commented:
True found the code for that

1-alex98 commented:
tested it look good
